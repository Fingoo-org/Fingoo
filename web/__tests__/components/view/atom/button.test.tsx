import { render } from '@testing-library/react';
import Button from '@/app/ui/components/view/atom/button/button';

describe('Button', () => {
  it('default 스타일의 버튼을 렌더링한다..', () => {
    const { container } = render(<Button color={'blue'}>버튼</Button>);

    expect(container).toMatchSnapshot();
  });

  it('primary 스타일의 버튼을 렌더링한다.', () => {
    const { container } = render(
      <Button color={'blue'} variant={'primary'}>
        버튼
      </Button>,
    );

    expect(container).toMatchSnapshot();
  });
  it('secondary 스타일의 버튼을 렌더링한다.', () => {
    const { container } = render(
      <Button color={'blue'} variant={'secondary'}>
        버튼
      </Button>,
    );

    expect(container).toMatchSnapshot();
  });
  it('light 스타일의 버튼을 렌더링한다.', () => {
    const { container } = render(
      <Button color={'blue'} variant={'light'}>
        버튼
      </Button>,
    );

    expect(container).toMatchSnapshot();
  });
});
