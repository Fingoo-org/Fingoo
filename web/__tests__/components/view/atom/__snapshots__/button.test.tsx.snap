// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button default 스타일의 버튼을 렌더링한다.. 1`] = `
<div>
  <button
    class="rounded-lg border px-4 py-2 text-sm bg-blue-500 border-blue-500 hover:bg-blue-600 text-white"
  >
    버튼
  </button>
</div>
`;

exports[`Button light 스타일의 버튼을 렌더링한다. 1`] = `
<div>
  <button
    class="text-sm text-blue-500 bg-transparent hover:text-blue-700"
  >
    버튼
  </button>
</div>
`;

exports[`Button primary 스타일의 버튼을 렌더링한다. 1`] = `
<div>
  <button
    class="rounded-lg border px-4 py-2 text-sm bg-blue-500 border-blue-500 hover:bg-blue-600 text-white"
  >
    버튼
  </button>
</div>
`;

exports[`Button secondary 스타일의 버튼을 렌더링한다. 1`] = `
<div>
  <button
    class="rounded-lg border px-4 py-2 text-sm bg-transparent border-blue-500 hover:bg-blue-500 hover:bg-opacity-20 dark:hover:bg-opacity-20 text-blue-500"
  >
    버튼
  </button>
</div>
`;
